Assets/static/scripts/login.js

```diff
@@ -1,10 +1,11 @@
-function send_login_request(username, password_hashed) {
+function send_login_request(username, password, cookie) {
     let packet = {
         "type": 2, // TYPE_AUTHENTICATE
         "isFinal": true,
         "object": {
             "username": username,
-            "password": password_hashed
+            "password": password,
+            "cookie-auth": cookie
```

Comment:
Would this field be present in the request if `cookie` is undefined?

If so the server might respond with a failed authentication regardless of the username / password field if i remember correctly.

Reply:
Just tested it, the field is not included when `cookie` is undefined.

Reply:
While it would still technically exist on the object, it will be ignored by `JSON.stringify` in the [packet constructor](https://github.com/Nathcat/PeopleCat-WebAppPrototype/blob/dac2a26a28ccedf72af47f568e95e553c151f27e/Assets/static/scripts/Packet.js#L45) and so not sent to the server!

[Take a look at this REPL if you're curious!](https://repljs.com/brooke_ec/hS3b52I41)